listenbrainz/webserver/views/test/test_user.py

```diff
@@ -11,11 +12,34 @@ def setUp(self):
         ServerTestCase.setUp(self)
         DatabaseTestCase.setUp(self)
         self.user = db_user.get_or_create('iliekcomputers')
+        self.weirduser = db_user.get_or_create('weird\\user name')
 
     def test_user_page(self):
         response = self.client.get(url_for('user.profile', user_name=self.user['musicbrainz_id']))
         self.assert200(response)
 
+    def test_scraper_username(self):
+        """ Tests that the username is correctly rendered in the last.fm importer """
+        response = self.client.get(
+            url_for('user.lastfmscraper', user_name=self.user['musicbrainz_id']),
+            query_string={
+                'user_token': self.user['auth_token'],
+                'lastfm_username': 'dummy',
+            }
+        )
+        self.assert200(response)
+        self.assertIn('var user_name = "{}";'.format(urllib.parse.quote_plus(self.user['musicbrainz_id'])), response.data.decode('utf-8'))
```

Comment:
It makes sense to use a plain string here as the first argument - `var user_name = "some\\\\name etc";` because we know exactly what it should be after escaping.

As this PR is about fixing users with spaces in their name, it'd be good to have this as a test too